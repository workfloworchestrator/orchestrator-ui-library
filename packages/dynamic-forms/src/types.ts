import React, { FormEventHandler } from 'react';
import { FieldValues, useForm } from 'react-hook-form';

import { z } from 'zod';

export type TDynamicFormFieldOptionsType = 'anyOf' | 'allOf';

export enum DfFieldFormats {
    DATE = 'date',
    HIDDEN = 'hidden',
    DATETIME = 'date-time',
    SKIP = 'skip',
    TIMESTAMP = 'timestamp',

    DEFAULT = 'text',
    DROPDOWN = 'dropdown',
    BOOLFIELD = 'boolfield',
    SELECT = 'select',
    MULTISELECT = 'multiselect',
    CHECKBOX = 'checkbox',
    RADIO = 'radio',
    SWITCH = 'switch',
    DATEPICKER = 'datepicker',
    NUMBER = 'number',
    LIST = 'list',
}

export enum DfFieldTypes {
    OPTGROUP = 'optGroup',
    SKIP = 'skip',
    LONG = 'long',
    HIDDEN = 'hidden',
    LABEL = 'label',
    SUMMARY = 'summary',
    ACCEPT = 'accept',
    NULL = 'null',

    DATE = 'date',
    DATETIME = 'date-time',
    OBJECT = 'object',
    STRING = 'string',
    ARRAY = 'array',
    BOOLEAN = 'boolean',
    NUMBER = 'number',
    TIMESTAMP = 'timestamp',
    LIST = 'list',
    DEFINED_LIST = 'defined-list',
    DEFINED_MULTI_LIST = 'defined-multi-list',
    TWO_OPTION_LIST = 'two-options-list',
}

export enum DFFieldReturnType {
    STRING = 'string',
    NUMBER = 'number',
    DATETIME = 'date-time',
}
export interface IValidationErrorDetails {
    detail: string;
    source: IDynamicFormApiValidationError[];
    mapped: {
        [fieldId: string]: {
            value: string;
            msg: string;
        };
    };
}

export interface IDynamicFormsContextProps {
    isLoading: boolean;
    isSending: boolean;
    isFullFilled: boolean;
    theReactHookForm: ReturnType<typeof useForm>;
    errorDetails?: IValidationErrorDetails;
    formData?: IDynamicForm;
    debugMode?: boolean;
    title?: string;
    sendLabel?: string;
    onCancel?: () => void;
    submitForm: FormEventHandler<HTMLFormElement>;
    resetForm: (e: MouseEvent) => void;
    successNotice?: React.ReactNode;
    headerComponent?: React.ReactNode;
    footerComponent?: React.ReactNode;
}

export type DynamicFormsMetaData = {
    [key: string | number]: DfFieldValue;
};

export type DfFormProvider = ({
    formKey,
    requestBody,
}: {
    formKey: string;
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    requestBody: any;
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
}) => Promise<Record<string, any>>;

export type DfLabelProvider = ({
    formKey,
    id,
}: {
    formKey: string;
    id?: string | null;
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
}) => Promise<Record<string, any>>;

export type DfDataProvider = () => Promise<IDynamicFormsLabels>;

export type CustomValidationRuleFn = (
    fieldConfig: IDynamicFormField,
    theReactHookForm?: ReturnType<typeof useForm>,
) => Zod.ZodTypeAny | undefined;

export interface IDynamicFormsContextConfig {
    // use custom method to provide data for the form. This overwrites data fetched from labels endpoint
    dataProvider: DfDataProvider;

    // use custom method for providing labels and data
    labelProvider: DfLabelProvider;

    // use a custom method for providing the form definition
    formProvider: DfFormProvider;

    // Extend field definitions
    fieldDetailProvider?: IFieldDefinitionProvider;

    // be able to refresh the provided data
    dataProviderCacheKey?: number;

    // whenever a fieldvalue changes, do something
    onFieldChangeHandler?: onFieldChangeHandlerFn;

    // provide custom validation rules for fields
    customValidationRules?: CustomValidationRuleFn;

    // This is a temp solution for 2 different Backend implementations of pydantic forms
    // we will remove this once the backends both work the same
    // Portal backend has a hack that is a leftover from previous implementation
    // So for only that backend, we should not send the data as a array
    tmp_pydanticFormsOriginalImplementation?: boolean;
}

export interface IDynamicFormsContextInitialProps {
    formKey: string;
    formIdKey?: string;
    title?: string;
    sendLabel?: string;
    metaData?: DynamicFormsMetaData;
    successNotice?: React.ReactNode;
    onSuccess?: (fieldValues: FieldValues) => void;
    onCancel?: () => void;
    children: (props: IDynamicFormsContextProps) => React.ReactNode;
    headerComponent?: React.ReactNode;
    footerComponent?: React.ReactNode;

    config: IDynamicFormsContextConfig;
}

export interface IFieldDefinitionProvider {
    [fieldId: string]: Partial<IDynamicFormField>;
}

export type onFieldChangeHandlerFn = (
    field: {
        name?: string;
        type?: string;
        value: DfFieldValue;
    },
    theReactHookForm: ReturnType<typeof useForm>,
) => void;

export interface IDynamicFormApiResponseDefEnum {}

export interface TJsonSchemaRef {
    $ref: string;
}

export interface TDynamicFormFieldAnyOfDef {
    items?: TJsonSchemaRef;
    type: 'null' | 'array';
}

export type ListFieldType =
    | DfFieldTypes.STRING
    | DfFieldTypes.DATE
    | DfFieldTypes.NUMBER
    | DfFieldTypes.NULL;

export interface ITDynamicFormFieldAnyOfResolvedItems {
    enum: string[];
    title: string;
    type: ListFieldType;
    maxLength?: number;
    minLength?: number;
    pattern?: string;
}

export interface TDynamicFormFieldAnyOfResolved {
    items?: ITDynamicFormFieldAnyOfResolvedItems;
    enum?: string[];
    format?: 'date' | 'date-time';
    type: ListFieldType;
    maxLength?: number;
    minLength?: number;
    pattern?: string;
}

export interface IDynamicFormApiResponseProperty {
    type: DfFieldTypes;
    anyOf?: TDynamicFormFieldAnyOfDef[];
    default?: string | null;
    title: string;
}

export interface IDynamicFormApiResponsePropertyResolved {
    type: DfFieldTypes;

    anyOf?: TDynamicFormFieldAnyOfResolved[];
    oneOf?: TDynamicFormFieldAnyOfResolved[];
    allOf?: TDynamicFormFieldAnyOfResolved[];

    items?: ITDynamicFormFieldAnyOfResolvedItems;
    enum?: string[];

    default?: string | null;
    title: string;
    format: DfFieldFormats;

    uniforms?: {
        disabled: boolean;
    };

    // validation props
    maxLength?: number;
    minLength?: number;
    pattern?: string;
}

export interface IDynamicFormApiResponseBase {
    title: string;
    description: string;
    additionalProperties: boolean;
    type: 'object';
    required?: string[];
    $defs?: {
        [definitionId: string]: {
            enum: string[];
            title: string;
            type: DfFieldTypes;
        };
    };
}

export interface IDynamicFormApiResponse extends IDynamicFormApiResponseBase {
    properties: {
        [propId: string]: IDynamicFormApiResponseProperty;
    };
}

export interface IDynamicFormApiErrorResponse {
    detail?: string;
    status: number;
    form: IDynamicFormApiResponse;
    success?: boolean;
    validation_errors: IDynamicFormApiValidationError[];
}

export interface IDynamicFormApiValidationError {
    input: string;
    loc: string[];
    msg: string;
    type: string;
    url: string; //"https://errors.pydantic.dev/2.4/v/extra_forbidden"
}

export interface IDynamicFormApiRefResolved
    extends IDynamicFormApiResponseBase {
    properties: {
        [propId: string]: IDynamicFormApiResponsePropertyResolved;
    };
}

export interface IDynamicFormsLabels {
    [key: string]: string[] | number[] | string | number | null;
}

export interface IDynamicFormsLabelResponse {
    labels: IDynamicFormsLabels;
}

export interface IDynamicFormFieldValidation {
    maxLength?: number;
    minLength?: number;
    pattern?: string;
    isNullable?: boolean;
}

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export type DfFieldValue = any;

export interface IDynamicFormFieldAttributes {
    disabled?: boolean;
}

export interface IDynamicFormField {
    id: string;
    title: string;
    description: string;
    type: DfFieldTypes;
    format: DfFieldFormats;
    options: string[];
    disabledOptions?: string[];
    default?: DfFieldValue;
    required: boolean;
    isEnumField: boolean;
    schemaField: IDynamicFormApiResponsePropertyResolved;
    validation: IDynamicFormFieldValidation;
    attributes: IDynamicFormFieldAttributes;
    validator?: FormZodValidationFn;
    FormElement?: FormElementComponent;
    matchedFieldResult?: DfFieldConfig;
}

export interface IDynamicFormFieldSection {
    id: string;
    title: string;
    fields: string[];
}

export enum IDynamicFormState {
    NEW = 'new',
    IN_PROGRESS = 'in-progress',
    FINISHED = 'finished',
}

export interface IDynamicForm {
    title: string;
    description: string;
    state: IDynamicFormState;
    fields: IDynamicFormField[];
    sections: IDynamicFormFieldSection[];
}

export type FormZodValidationFn = (field: IDynamicFormField) => z.ZodTypeAny;
export type FormElementComponent = (props: IDFInputFieldProps) => JSX.Element;

export interface FormComponent {
    Element: FormElementComponent;
    validator?: FormZodValidationFn;
}

export interface DfFieldConfig {
    id: string;
    Component: FormComponent;
    matcher?: (field: IDynamicFormField) => boolean;
    preventColRender?: boolean;
}

export type DfFieldsConfig = DfFieldConfig[];

export interface IDFInputFieldProps {
    field: IDynamicFormField;
}

export interface IDFZodValidationPresets {
    [type: string]: FormZodValidationFn;
}
