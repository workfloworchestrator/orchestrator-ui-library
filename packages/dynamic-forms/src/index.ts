import DynamicForm from "./components/form/ContextWrap"

export default DynamicForm
