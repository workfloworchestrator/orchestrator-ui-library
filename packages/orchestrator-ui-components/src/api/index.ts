/*
 * Copyright 2019-2023 SURF.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *         http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 */
import { AxiosInstance } from 'axios';

import {
    ImsPort,
    IpBlock,
    IpPrefix,
    NodeSubscription,
} from '../components/WfoForms/formFields/surf/types';
import { ProductDefinition } from '../types';
import { getAxiosInstance } from './axios';

const FORMS_ENDPOINT = 'surf/forms/';
const PROCESS_ENDPOINT = 'processes/';
const PRODUCTS_ENDPOINT = 'products/';

export class BaseApiClient {
    private _axiosInstance: AxiosInstance;

    constructor(apiPath: string) {
        this._axiosInstance = getAxiosInstance(apiPath);
    }

    axiosFetch = <R = object>(
        path: string,
        options = {},
        // eslint-disable-next-line @typescript-eslint/no-unused-vars
        headers = {},
        showErrorDialog = true,
        // eslint-disable-next-line @typescript-eslint/no-unused-vars
        result = true,
    ): Promise<R> => {
        // preset the config with the relative URL and a GET type.
        // presets can be overridden with `options`.
        return this._axiosInstance({ url: path, method: 'GET', ...options })
            .then((res) => res.data)
            .catch((err) => {
                if (showErrorDialog) {
                    setTimeout(() => {
                        throw err;
                    }, 250);
                }
                throw err;
            });
    };

    catchErrorStatus = <T>(
        promise: Promise<unknown>,
        status: number,
        callback: (json: T) => void,
    ) => {
        return promise.catch((err) => {
            if (err.response && err.response.status === status) {
                callback(err.response.data);
            } else {
                throw err;
            }
        });
    };

    fetchJson = <R = object>(
        path: string,
        options = {},
        headers = {},
        showErrorDialog = true,
        result = true,
    ): Promise<R> => {
        return this.axiosFetch(path, options, headers, showErrorDialog, result);
    };

    fetchJsonWithCustomErrorHandling = <R = object>(
        path: string,
    ): Promise<R> => {
        return this.fetchJson(path, {}, {}, false, true);
    };

    postPutJson = <R = object>(
        path: string,
        processInput: object,
        method: string,
        showErrorDialog = true,
        result = true,
    ): Promise<R> => {
        return this.axiosFetch(
            path,
            { method: method, data: processInput },
            {},
            showErrorDialog,
            result,
        );
    };
}

abstract class ApiClientInterface extends BaseApiClient {
    abstract cimStartForm: (
        formKey: string,
        userInputs: object[],
    ) => Promise<object>;
}

export class ApiClient extends ApiClientInterface {
    startProcess = (
        workflowName: string,
        processInput: object,
    ): Promise<unknown> => {
        return this.postPutJson<unknown>(
            `${PROCESS_ENDPOINT}${workflowName}`,
            processInput,
            'post',
            false,
            true,
        );
    };

    resumeProcess = (processId: string, userInput: object[]) => {
        return this.postPutJson(
            `processes/${processId}/resume`,
            userInput,
            'put',
            false,
            false,
        );
    };

    products = (): Promise<ProductDefinition[]> => {
        return this.fetchJson<ProductDefinition[]>(PRODUCTS_ENDPOINT);
    };
    productById = (productId: string): Promise<ProductDefinition> => {
        return this.fetchJson(`${PRODUCTS_ENDPOINT}${productId}`);
    };
    cimStartForm = (
        formKey: string,
        userInputs: object[],
    ): Promise<{ id: string }> => {
        return this.postPutJson(
            `${FORMS_ENDPOINT}${formKey}`,
            userInputs,
            'post',
            false,
            true,
        );
    };
    prefix_filters = (): Promise<IpPrefix[]> => {
        return this.fetchJson('surf/ipam/prefix_filters');
    };
    ip_blocks = (parentPrefix: number): Promise<IpBlock[]> => {
        return this.fetchJson('surf/ipam/ip_blocks/' + parentPrefix);
    };
    free_subnets = (
        subnet: string,
        netmask: number,
        prefixlen: number,
    ): Promise<string[]> => {
        return this.fetchJson(
            'surf/ipam/free_subnets/' +
                subnet +
                '/' +
                netmask +
                '/' +
                prefixlen,
        );
    };
    getFreePortsByNodeSubscriptionIdAndSpeed = (
        nodeSubscriptionId: string,
        interfaceSpeed: number,
        mode: string,
    ): Promise<ImsPort[]> => {
        return this.fetchJson(
            `surf/ims/free_ports/${nodeSubscriptionId}/${interfaceSpeed}/${mode}`,
        );
    };

    // legacy : for imsPort selector
    subscriptions = (
        tagList: string[] = [],
        statusList: string[] = [],
        productList: string[] = [],
    ): Promise<NodeSubscription[]> => {
        const filters = [];

        if (tagList.length)
            filters.push(`tags,${encodeURIComponent(tagList.join('-'))}`);
        if (statusList.length)
            filters.push(
                `statuses,${encodeURIComponent(statusList.join('-'))}`,
            );
        if (productList.length)
            filters.push(
                `products,${encodeURIComponent(productList.join('-'))}`,
            );

        const params = new URLSearchParams();
        if (filters.length) params.set('filter', filters.join(','));

        return this.fetchJson(
            `subscriptions/${filters.length ? '?' : ''}${params.toString()}`,
        );
    };

    nodeSubscriptions = (
        statusList: string[] = [],
    ): Promise<NodeSubscription[]> => {
        return this.subscriptions(['Node'], statusList);
    };
}

export function getApiClient(apiEndPoint: string) {
    return new ApiClient(apiEndPoint);
}
