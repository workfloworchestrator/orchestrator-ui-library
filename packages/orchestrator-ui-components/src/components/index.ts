export * from './WfoBadges';
export * from './WfoPageTemplate/WfoBreadcrumbs';
export * from './WfoFilterTabs';
export * from './WfoKeyValueTable';
export * from './WfoPageTemplate';
export * from './WfoSearchBar';
export * from './WfoSettingsModal';
export * from './WfoSettingsPage';
export * from './WfoStartPage';
export * from './WfoSubscription';
export * from './WfoTable';
export * from './WfoTimeline';
export * from './WfoLoading';
export * from './WfoToastsList';
export * from './WfoForms';
export * from './WfoStartTaskButtonComboBox';
