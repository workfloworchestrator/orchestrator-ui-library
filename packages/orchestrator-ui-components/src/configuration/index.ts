export * from './constants';
export * from './policy-resources';
