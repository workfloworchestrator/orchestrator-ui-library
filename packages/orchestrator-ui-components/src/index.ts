export * from './lib/components';

export * from './lib/icons';

export * from './lib/theme';

export * from './lib/utils';

export * from './lib/contexts';

export * from './lib/types';

export * from './lib/hooks';
