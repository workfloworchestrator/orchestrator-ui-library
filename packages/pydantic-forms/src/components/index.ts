export * from './forms';
export * from './formFields';
