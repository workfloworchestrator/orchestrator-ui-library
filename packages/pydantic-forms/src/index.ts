export * from './components/index';
export * from './contexts/index';
export * from './messages/index';
export * from './types/index';
