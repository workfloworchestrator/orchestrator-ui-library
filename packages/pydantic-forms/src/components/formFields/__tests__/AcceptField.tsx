import createContext from "lib/uniforms-surfnet/__tests__/_createContext";
import mount from "lib/uniforms-surfnet/__tests__/_mount";
import { AcceptField } from "lib/uniforms-surfnet/src";
/*
 * Copyright 2019-2023 SURF.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *         http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 */
import React from "react";

const TEST_ACCEPT_DATA = [
    ["label", "label"],
    ["info", "info"],
    ["warning", "warning"],
    ["margin", "margin"],
    ["value", "value"],
    ["http://example.com", "url"],
    ["checkbox1", "checkbox"],
    ["label_with_context", "label", { foo: "bar" }],
    ["sub_checkbox", ">checkbox"],
    ["sub_checkbox2", ">checkbox"],
    ["optional_checkbox", "checkbox?"],
    ["skip_checkbox", "skip"],
];

describe("<AcceptField>", () => {
    test("<AcceptField> - legacy renders an input", () => {
        const element = <AcceptField name="x" />;
        const wrapper = mount(element, createContext({ x: { type: String } }, undefined, "acceptfield-legacy"));
        expect(wrapper.find("input")).toHaveLength(1);
        expect(wrapper.render()).toMatchSnapshot();
    });

    test("<AcceptField> - legacy renders a input which correctly reacts on change", () => {
        const onChange = jest.fn();

        const element = <AcceptField name="x" />;
        const wrapper = mount(element, createContext({ x: { type: String } }, { onChange }));

        expect(wrapper.find("input[name='x']")).toHaveLength(1);
        expect(wrapper.find("input[name='x']").simulate("change", { target: { checked: true } })).toBeTruthy();
        expect(onChange).toHaveBeenLastCalledWith("x", "ACCEPTED");
        expect(wrapper.find("input[name='x']").simulate("change", { target: { checked: false } })).toBeTruthy();
        expect(onChange).toHaveBeenLastCalledWith("x", "INCOMPLETE");
    });

    test("<AcceptField> - renders inputs", () => {
        const element = <AcceptField name="x" />;
        const wrapper = mount(
            element,
            createContext(
                {
                    x: {
                        type: String,
                        uniforms: {
                            data: TEST_ACCEPT_DATA,
                        },
                    },
                },
                undefined,
                "acceptfield-render"
            )
        );
        expect(wrapper.find("input")).toHaveLength(6);
        expect(wrapper.render()).toMatchSnapshot();
    });
    test("<AcceptField> - renders a input which correctly reacts on change", () => {
        const onChange = jest.fn();

        const element = <AcceptField name="x" />;
        const wrapper = mount(
            element,
            createContext(
                {
                    x: {
                        type: String,
                        uniforms: {
                            data: TEST_ACCEPT_DATA,
                        },
                    },
                },
                { onChange }
            )
        );

        expect(wrapper.find("input[name='checkbox16']")).toHaveLength(1);
        expect(wrapper.find("input[name='sub_checkbox8']")).toHaveLength(1);
        expect(wrapper.find("input[name='sub_checkbox29']")).toHaveLength(1);
        expect(wrapper.find("input[name='optional_checkbox10']")).toHaveLength(1);
        expect(wrapper.find("input[name='skip_checkbox']")).toHaveLength(1);

        expect(wrapper.find("input[name='checkbox16']").simulate("change", { target: { checked: true } })).toBeTruthy();
        expect(onChange).toHaveBeenLastCalledWith("x", "INCOMPLETE");

        expect(
            wrapper.find("input[name='sub_checkbox8']").simulate("change", { target: { checked: true } })
        ).toBeTruthy();
        expect(onChange).toHaveBeenLastCalledWith("x", "INCOMPLETE");

        expect(
            wrapper.find("input[name='optional_checkbox10']").simulate("change", { target: { checked: true } })
        ).toBeTruthy();
        expect(onChange).toHaveBeenLastCalledWith("x", "INCOMPLETE");

        expect(
            wrapper.find("input[name='sub_checkbox29']").simulate("change", { target: { checked: true } })
        ).toBeTruthy();
        expect(onChange).toHaveBeenLastCalledWith("x", "ACCEPTED");

        expect(
            wrapper.find("input[name='optional_checkbox10']").simulate("change", { target: { checked: false } })
        ).toBeTruthy();
        expect(onChange).toHaveBeenLastCalledWith("x", "ACCEPTED");

        expect(
            wrapper.find("input[name='skip_checkbox']").simulate("change", { target: { checked: true } })
        ).toBeTruthy();
        expect(onChange).toHaveBeenLastCalledWith("x", "SKIPPED");

        expect(
            wrapper.find("input[name='skip_checkbox']").simulate("change", { target: { checked: false } })
        ).toBeTruthy();
        expect(onChange).toHaveBeenLastCalledWith("x", "INCOMPLETE");

        expect(wrapper.find("input[name='checkbox16']").simulate("change", { target: { checked: true } })).toBeTruthy();
        expect(onChange).toHaveBeenLastCalledWith("x", "INCOMPLETE");
    });
});
