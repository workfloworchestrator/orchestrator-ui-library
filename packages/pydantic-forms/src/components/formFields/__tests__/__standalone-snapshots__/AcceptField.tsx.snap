// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AcceptField> <AcceptField> - legacy renders an input 1`] = `
LoadedCheerio {
  "0": <div
    class="euiFlexItem css-iqegkc-EuiFlexItem"
  >
    <section
      class=" accept-field"
      id="acceptfield-legacy-0000"
      required=""
    >
      <label
        class="euiFormLabel euiFormRow__label"
      >
        forms.fields.x
      </label>
      <div
        class="euiText euiText--medium"
      >
        forms.fields.x_info
      </div>
      <div
        class="euiCheckbox"
      >
        <input
          class="euiCheckbox__input"
          id="x"
          name="x"
          type="checkbox"
        />
        <div
          class="euiCheckbox__square"
        />
        <label
          class="euiCheckbox__label"
          for="x"
        >
          forms.fields.x
        </label>
      </div>
    </section>
  </div>,
  "_root": LoadedCheerio {
    "0": Document {
      "children": Array [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`<AcceptField> <AcceptField> - renders inputs 1`] = `
LoadedCheerio {
  "0": <div
    class="euiFlexItem css-iqegkc-EuiFlexItem"
  >
    <section
      class=" accept-field"
      id="acceptfield-render-0000"
      required=""
    >
      <label
        class="euiFormLabel euiFormRow__label"
      >
        forms.fields.x_accept.label
      </label>
      <div
        class="euiText euiText--medium"
      >
        forms.fields.x_accept.info
      </div>
      <label
        class="euiFormLabel euiFormRow__label warning"
      >
        forms.fields.x_accept.warning
      </label>
      <br />
      <div>
        <input
          disabled=""
          value="value"
        />
      </div>
      <div>
        <a
          href="http://example.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          http://example.com
        </a>
      </div>
      <div
        class="euiCheckbox"
      >
        <input
          class="euiCheckbox__input"
          id="checkbox16"
          name="checkbox16"
          type="checkbox"
        />
        <div
          class="euiCheckbox__square"
        />
        <label
          class="euiCheckbox__label"
          for="checkbox16"
        >
          forms.fields.x_accept.checkbox1
        </label>
      </div>
      <label
        class="euiFormLabel euiFormRow__label"
      >
        forms.fields.x_accept.label_with_context
      </label>
      <div
        class="euiCheckbox level_2"
      >
        <input
          class="euiCheckbox__input"
          id="sub_checkbox8"
          name="sub_checkbox8"
          type="checkbox"
        />
        <div
          class="euiCheckbox__square"
        />
        <label
          class="euiCheckbox__label"
          for="sub_checkbox8"
        >
          forms.fields.x_accept.sub_checkbox
        </label>
      </div>
      <div
        class="euiCheckbox level_2"
      >
        <input
          class="euiCheckbox__input"
          id="sub_checkbox29"
          name="sub_checkbox29"
          type="checkbox"
        />
        <div
          class="euiCheckbox__square"
        />
        <label
          class="euiCheckbox__label"
          for="sub_checkbox29"
        >
          forms.fields.x_accept.sub_checkbox2
        </label>
      </div>
      <div
        class="euiCheckbox"
      >
        <input
          class="euiCheckbox__input"
          id="optional_checkbox10"
          name="optional_checkbox10"
          type="checkbox"
        />
        <div
          class="euiCheckbox__square"
        />
        <label
          class="euiCheckbox__label"
          for="optional_checkbox10"
        >
          forms.fields.x_accept.optional_checkbox
        </label>
      </div>
      <div
        class="euiCheckbox skip"
      >
        <input
          class="euiCheckbox__input"
          id="skip_checkbox"
          name="skip_checkbox"
          type="checkbox"
        />
        <div
          class="euiCheckbox__square"
        />
        <label
          class="euiCheckbox__label"
          for="skip_checkbox"
        >
          forms.fields.x_accept.skip_checkbox
        </label>
      </div>
    </section>
  </div>,
  "_root": LoadedCheerio {
    "0": Document {
      "children": Array [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
